java-frontend/src/main/java/org/sonar/java/ProgressMonitor.java

```diff
@@ -39,8 +40,19 @@
   private boolean unknownTotalWork = false;
   private int processedWork = 0;
 
+  /**
+   * The report loop can not rely only on Thread.interrupted() to end, according to
+   * interrupted() javadoc, a thread interruption can be ignored because a thread was
+   * not alive at the time of the interrupt. This could happen if done() is being called
+   * before ProgressReport's thread becomes alive.
+   * So this boolean flag ensures that ProgressReport never enter an infinite loop when
+   * Thread.interrupted() failed to be set to true.
+   */
```

Comment:
You should replace 2 times `ProgressReport` with `ProgressMonitor` if you don't want that someone notices your copy and paste :wink:

Reply:
😅